src/content/content_type.rs

```diff
@@ -0,0 +1,140 @@
+use std::{convert::TryInto, str::FromStr};
+
+use crate::headers::{HeaderName, HeaderValue, Headers, CONTENT_TYPE};
+use crate::Mime;
+
+/// Indicate the media type of the resource.
+///
+/// [MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
+///
+/// # Specifications
+///
+/// - [RFC 7231, section 3.1.1.5: Content-Type](https://tools.ietf.org/html/rfc7231#section-3.1.1.5)
+/// - [RFC 7233, section 4.1: Content-Type in multipart](https://tools.ietf.org/html/rfc7233#section-4.1)
+///
+/// # Examples
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::content::ContentType;
+/// use http_types::{Response, Mime};
+/// use std::str::FromStr;
+///
+/// let content_type = ContentType::new(Mime::from_str("text/*")?);
```

Comment:
Can this be?:
```suggestion
/// let content_type = ContentType::new("text/*");
```

Or is the issue that `new()` here unwraps?